api/auth/cert/cert.go

```diff
@@ -0,0 +1,163 @@
+package cert
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/hashicorp/go-rootcerts"
+	"github.com/hashicorp/vault/api"
+)
+
+type CertAuth struct {
+	role               string
+	caCert             string
+	caCertBytes        []byte
+	clientCert         string
+	clientKey          string
+	insecureSkipVerify bool
+}
+
+var _ api.AuthMethod = (*CertAuth)(nil)
+
+type LoginOption func(a *CertAuth) error
+
+// NewCertAuth initializes a new cert auth method interface to be
+// passed as a parameter to the client.Auth().Login method.
+//
+// Supported options: WithClientCertAndKey, WithInsecure
+//
+// https://developer.hashicorp.com/vault/api-docs/auth/cert#login-with-tls-certificate-method
+func NewCertAuth(roleName string, opts ...LoginOption) (*CertAuth, error) {
+	a := &CertAuth{
+		role: roleName,
+	}
+
+	for _, opt := range opts {
+		err := opt(a)
+		if err != nil {
+			return nil, fmt.Errorf("error with login option: %w", err)
+		}
+	}
+
+	return a, nil
+}
+
+// Login sets up the required request body for the cert auth method's /login
+// endpoint, and performs a write to it.
+// It adds the client cert and key to the request.
+func (a *CertAuth) Login(ctx context.Context, client *api.Client) (*api.Secret, error) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	c, err := a.httpClient()
+	if err != nil {
+		return nil, err
+	}
+
+	reqBody, err := json.Marshal(map[string]interface{}{
+		"name": a.role,
```

Comment:
```suggestion
		"role": a.role,
```
Shouldn't this be `role`?

Reply:
Sounds good!